### PR TITLE
Add support for data modification before rendering and variables

### PR DESCRIPTION
**Pull request by ThaNarie**
* Update muban-core to have support for `onData`
* Exclude data files starting with a `_` as a page
* Add `_variables.yaml` to contain global variables
* Add variable replacement in `bootstrap.dev` (during dev) and `getPages` (during build)
* Add support for `js` as data files
* Add documentation on data files

### CODE REVIEW

File: src/data/home.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      name: 'header',
+      data: {},
+    },
+    {
+      name: 'paragraph',
+      data: 'import!../app/component/block/paragraph/data.js',
+    },
+    {
+      name: 'two-col',
+      data: 'import!../app/component/block/two-col/data.js',
```
Comment (pigeonfresh) [inline src/data/home.js]: The Two Columns block imports a JS file where there is only a yaml file available

Comment (pigeonfresh) [reply src/data/home.js]: My bad!

File: docs/data-files.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
+# Data
+
+The data files, located at `src/data/`, provide the content and the structure of your pages.
+Every page has it's own data file, which can be either `yaml`, `json` or `js`.
+Since `yaml` is less verbose, and can better handle multiline content, we've chosen
+that as the default. `js` can be useful for more dynamic data, when creating loops or needing
+something from `process.env`.
+
+To reader a page, the `src/app/component/layout/app/app.hbs` is provided with the data for the page,
```
Comment (larsvanbraam) [inline docs/data-files.md]:
```suggestion
To render a page, the `src/app/component/layout/app/app.hbs` is provided with the data for the page,
```

File: docs/data-files.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+_**Note**: data files that should not end up as a page, should start with a `_`, so they are skipped._
+
+## Modify data & variables
+
+If you want to have a bit more control over your data, you are able to change the data
+right before it gets rendered. During dev this can be done using the `onData` callback
+that is passed to the dev `bootstrap` as an option. During build you can change the data
+in `build-html.js` for each page before rendering it to HTML.
+
+By default, Muban already allow you to use variables in your data files, that will be
```
Comment (larsvanbraam) [inline docs/data-files.md]:
```suggestion
By default, Muban already allows you to use variables in your data files, that will be
```

**Pull request merged**